### PR TITLE
COMMERCE-3408 Product Card migration, with new components and product rendering strategies refactor

### PR DESCRIPTION
**Pull request by gianmarcobrunialti**

### CODE REVIEW

**Comment by gianmarcobrunialti** [comment]
ci:test:sf

**Comment by gianmarcobrunialti** [comment]
ci:test:relevant

**Comment by gianmarcobrunialti** [comment]
ci:stop

**Comment by ivicac** [comment]
ci:test:relevant

**Comment by ivicac** [comment]
ci:test:relevant

**Comment by gianmarcobrunialti** [comment]
ci:test:sf

**Comment by gianmarcobrunialti** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/AddToOrderTag.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+	public boolean isWillUpdate() {
+		return _willUpdate;
+	}
+
+	@Override
+	public void setAttributes(HttpServletRequest httpServletRequest) {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		if (Validator.isNull(_spritemap)) {
```
Comment (ivicac) [inline modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/AddToOrderTag.java:83]: Move where it is used

File: modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/AddToWishListTag.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+	public boolean isLarge() {
+		return _large;
+	}
+
+	@Override
+	public void setAttributes(HttpServletRequest httpServletRequest) {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		if (Validator.isNull(_spritemap)) {
```
Comment (ivicac) [inline modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/AddToWishListTag.java:59]: Move where it is used

File: modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/ProductCardTag.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		setNamespacedAttribute(
+			httpServletRequest, "displayProductInformation",
+			_displayProductInformation);
+		setNamespacedAttribute(
+			httpServletRequest, "elementClasses", _elementClasses);
+	}
+
+	public void setDisplayProductInformation(HashMap<String, Object> context) {
```
Comment (ivicac) [inline modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/ProductCardTag.java:49]: Let's introduce DisplayProductInformation class that will replace HashMap

File: modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/availability_label/page.jsp [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	<span class="label-item label-item-expand">
+		<%= LanguageUtil.get(request, availabilityLabel) %>
+	</span>
+</span>
+
+<%
+if (GetterUtil.getBoolean(willUpdate)) {
+%>
+
+	<aui:script require="commerce-frontend-js/utilities/eventsDefinitions as Events">
```
Comment (ivicac) [inline modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/availability_label/page.jsp:49]: Move this function outside to be able to unit test it

File: modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/product_card/add_to_cart.jsp [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+boolean inCart = (boolean)displayProductInformation.get("inCart");
+long orderId = (long)displayProductInformation.get("orderId");
+long skuId = (long)displayProductInformation.get("skuId");
+int stockQuantity = (int)displayProductInformation.get("stockQuantity");
+
+String randomNamespace = PortalUtil.generateRandomKey(request, "taglib") + StringPool.UNDERLINE;
+%>
+
+<%
+if (skuId != 0) {
```
Comment (ivicac) [inline modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/product_card/add_to_cart.jsp:32]: Please, invert the logic

**Review by ivicac — CHANGES_REQUESTED** [review]

**Comment by ivicac** [comment]
@dsitu can you please check this PR, so all necessary changes are done before fixing tests

File: modules/apps/commerce/commerce-frontend-js/test/dev/components/index.js [hunk 6]
```diff
@@ -18,6 +18,11 @@ module.exports = [
 		name: 'Add To Cart',
 		page: 'add-to-cart.html',
 	},
+	{
+		entry: 'AddToWishList',
+		name: 'Add To WishList',
```
Comment (dsitu) [inline modules/apps/commerce/commerce-frontend-js/test/dev/components/index.js:23]:
```suggestion
		name: 'Add To Wish List',
```

File: modules/apps/commerce/commerce-product-content-web/src/main/resources/META-INF/resources/product_detail/render/view.jsp [hunk 7]
```diff
@@ -25,7 +25,25 @@ CPSku cpSku = cpContentHelper.getDefaultCPSku(cpCatalogEntry);
...
+
+long commerceAccountId = (long)displayProductInformation.get("commerceAccountId");
+String currencyCode = (String)displayProductInformation.get("currencyCode");
+long channelId = (long)displayProductInformation.get("channelId");
+boolean inCart = (boolean)displayProductInformation.get("inCart");
+boolean inWishList = (boolean)displayProductInformation.get("inWishList");
+boolean lowStock = (boolean)displayProductInformation.get("lowStock");
+long commerceOrderId = (long)displayProductInformation.get("commerceOrderId");
+int stockQuantity = (int)displayProductInformation.get("stockQuantity");
+String spritemap = (String)displayProductInformation.get("spritemap");
```
Comment (dsitu) [inline modules/apps/commerce/commerce-product-content-web/src/main/resources/META-INF/resources/product_detail/render/view.jsp:38]: Lets just call is `orderId` and sort by alphabetical order.
```suggestion
long channelId = (long)displayProductInformation.get("channelId");
long commerceAccountId = (long)displayProductInformation.get("commerceAccountId");
String currencyCode = (String)displayProductInformation.get("currencyCode");
boolean inCart = (boolean)displayProductInformation.get("inCart");
boolean inWishList = (boolean)displayProductInformation.get("inWishList");
boolean lowStock = (boolean)displayProductInformation.get("lowStock");
long orderId = (long)displayProductInformation.get("commerceOrderId");
String spritemap = (String)displayProductInformation.get("spritemap");
int stockQuantity = (int)displayProductInformation.get("stockQuantity");
```

**Review by dsitu — CHANGES_REQUESTED** [review]

**Comment by gianmarcobrunialti** [comment]
ci:test:sf

**Comment by gianmarcobrunialti** [comment]
@marco-leo As we discussed offline, it would be nice to understand the logics concerning the enabling/disabling the purchasing/add-to-cart interactions, as they won't be solely based on `stockQuantity` or product availability. Can we cover this up before the final review?

@ivicac All of the other comments are now covered up 👍

**Comment by gianmarcobrunialti** [comment]
ci:test:sf

**Comment by gianmarcobrunialti** [comment]
ci:test:sf

**Comment by gianmarcobrunialti** [comment]
ci:test:relevant

**Comment by ivicac** [comment]
ci:forward

**Comment by ivicac** [comment]
ci:forward:stop

**Comment by ivicac** [comment]
ci:forward

**Pull request closed without merging**